action/actctx.go

```diff
@@ -62,27 +59,19 @@ func (act *AbstractAction) SetGasPrice(val *big.Int) {
 func (act *AbstractAction) BasicActionSize() uint32 {
 	// VersionSizeInBytes + NonceSizeInBytes + GasSizeInBytes
 	size := 4 + 8 + 8
-	if act.srcPubkey != nil {
-		size += len(act.srcPubkey.Bytes())
-	}
 	if act.gasPrice != nil && len(act.gasPrice.Bytes()) > 0 {
 		size += len(act.gasPrice.Bytes())
 	}
 
 	return uint32(size)
 }
 
-// SetEnvelopeContext sets the SealedEnvelope context to action context.
-func (act *AbstractAction) SetEnvelopeContext(selp SealedEnvelope) {
+// SetAbstractAction sets the struct to input
+func (act *AbstractAction) SetAbstractAction(ab AbstractAction) {
```

Comment:
now we don't need pass in the entire sealedEnvelope struct, only need AbstractAction